webui/react/src/stores/determinedInfo.tsx

```diff
@@ -17,21 +51,31 @@ export const initInfo: DeterminedInfo = {
   version: process.env.VERSION || '',
 };
 
-const info = observable<Loadable<DeterminedInfo>>(NotLoaded);
+class DeterminedStore extends PollingStore {
+  #info: WritableObservable<Loadable<DeterminedInfo>> = observable(NotLoaded);
+
+  public readonly loadableInfo = this.#info.readOnly();
 
-export const fetchDeterminedInfo = async (canceler: AbortController): Promise<void> => {
-  try {
-    const response = await getInfo({ signal: canceler.signal });
-    info.set(Loaded(response));
-  } catch (e) {
-    info.update((prevInfo) => {
-      const info = Loadable.getOrElse(initInfo, prevInfo);
+  public readonly info = this.#info.select((info) => Loadable.getOrElse(initInfo, info));
+
+  public readonly isServerReachable = this.#info.select((info) => {
+    return Loadable.match(info, {
+      Loaded: (info) => !!info.clusterId,
+      NotLoaded: () => false,
+    });
+  });
+
+  protected async poll() {
```

Comment:
do we need polling here?

Reply:
To clarify, are you asking whether we need polling for info store at all or whether the polling is needed at that particular location? If it's the former, the `App.tsx` is currently polling every 600,000 ms (10min) and I didn't want to change that particular behavior for now. I think it's a good question and the one area where I think the polling is useful is when the cluster upgrades and the version changes. This polling would allow existing open Web UIs to detect version mismatch and show that notification.

Reply:
ah ok for some reason I remembered this as being like `auth` (fetched only once on startup in `App.tsx`). Keeping the existing polling sounds fine.